### PR TITLE
De-linting for Arm_Control_System Branch

### PR DESCRIPTION
**Pull request by Tzanccc**
De-linting the arm control system to pass clang tests. Refactoring wr_control_drive_arm as well as removing unused code.

### CODE REVIEW

File: src/wr_control_drive_arm/src/ArmControlSystem.cpp [hunk 1]
```diff
@@ -15,24 +15,48 @@
...
+constexpr float TIMER_CALLBACK_DURATION = 1.0 / 50.0;
+
 /**
  * @brief Defines space for all ArmMotor references
  */
-const int numMotors = 6;
-ArmMotor *motors[numMotors];
+// constexpr int NUM_MOTORS = 6;
+// std::array<std::unique_ptr<ArmMotor>, NUM_MOTORS> motors;
+// ArmMotor *motors[numMotors];
```
Comment (bennowotny) [inline src/wr_control_drive_arm/src/ArmControlSystem.cpp:52]: If not used, can be removed.

File: src/wr_control_drive_arm/src/ArmControlSystem.cpp [hunk 2]
```diff
@@ -225,43 +204,41 @@ int main(int argc, char** argv)
   pn.getParam("encoder_parameters", encParams);
 
   // Initialize all motors with their MoveIt name, WRoboclaw initialization, and reference to the current node
-  motors[0] = new ArmMotor("elbow", 1, 0, static_cast<int>(encParams[0]["counts_per_rotation"]), static_cast<int>(encParams[0]["offset"]), n);
-  motors[1] = new ArmMotor("forearm_roll", 1, 1, static_cast<int>(encParams[1]["counts_per_rotation"]), static_cast<int>(encParams[1]["offset"]), n);
-  motors[2] = new ArmMotor("shoulder", 0, 1, static_cast<int>(encParams[2]["counts_per_rotation"]), static_cast<int>(encParams[2]["offset"]), n);
-  motors[3] = new ArmMotor("turntable", 0, 0, static_cast<int>(encParams[3]["counts_per_rotation"]), static_cast<int>(encParams[3]["offset"]), n);
-  motors[4] = new ArmMotor("wrist_pitch", 2, 0, static_cast<int>(encParams[4]["counts_per_rotation"]), static_cast<int>(encParams[4]["offset"]), n);
-  motors[5] = new ArmMotor("wrist_roll", 2, 1, static_cast<int>(encParams[5]["counts_per_rotation"]), static_cast<int>(encParams[5]["offset"]), n);
+  // TODO: fix
```
Comment (bennowotny) [inline src/wr_control_drive_arm/src/ArmControlSystem.cpp:207]: Fixed, so can be removed.

File: src/wr_control_drive_arm/src/ArmControlSystem.cpp [hunk 3]
```diff
@@ -225,43 +204,41 @@ int main(int argc, char** argv)
...
   // Initialize all Joints
-  joints[0] = new SimpleJoint(motors[0], &n);
-  joints[1] = new SimpleJoint(motors[1], &n);
-  joints[2] = new SimpleJoint(motors[2], &n);
-  joints[3] = new SimpleJoint(motors[3], &n);
-  joints[4] = new SimpleJoint(motors[4], &n);
-  DifferentialJoint* temp = new DifferentialJoint(motors[4], motors[5], &n);
-  temp->configVelocityHandshake("/control/arm/5/roll", "/control/arm/5/pitch", "/control/arm/20/", "/control/arm/21/");
-
+  // TODO: fix
```
Comment (bennowotny) [inline src/wr_control_drive_arm/src/ArmControlSystem.cpp:217]: Fixed, so can be removed.

**Review by bennowotny — CHANGES_REQUESTED** [review]
Some old, extraneous code to removed before merging.

**Review by Tzanccc — COMMENTED** [review]
Old code removed

**Review by bennowotny — APPROVED** [review]
Ready to go, will be merged.

**Pull request merged**